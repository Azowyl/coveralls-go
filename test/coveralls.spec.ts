import Coveralls from '../src/coveralls';
import Environment from '../src/environment';
import axios, { AxiosRequestConfig } from 'axios';
import FormData from 'form-data';

jest.mock('axios');
jest.mock('js-md5', () => ({
    digest: jest.fn().mockReturnValue('sourceDigest'),
}));

describe('Coveralls', () => {
    let coveralls: Coveralls;
    const axiosMock = axios as unknown as jest.Mock;

    beforeEach(() => {
        coveralls = new Coveralls();
    });

    const itMakesRequestWithCorrectParams = (): void =>
        it('makes request to coveralls with correct params', async () => {
            await coveralls.submitFromLcov(
                require.resolve('./fixtures/repo/coverage/lcov.info')
            );

            const calledData = (
                axiosMock.mock.calls[0] as object[]
            )[0] as AxiosRequestConfig;

            expect(
                (calledData.data as FormData)
                    .getBuffer()
                    .toString()
                    .match(/{.*}/)
            ).toMatchSnapshot();
        });

    describe('submitFromLcov', () => {
        describe('with PR circleci environment', () => {
            beforeEach(() => {
                Environment.CIRCLECI = true;
                Environment.CIRCLE_BUILD_NUM = '1';
                Environment.CIRCLE_COMMIT_SHA =
                    '0d15c65a92b7c3642883a528da8dd4841accde21';
                Environment.CIRCLE_PULL_REQUEST_ID = '123';
            });

            itMakesRequestWithCorrectParams();
        });

        describe('with no PR circleci environment', () => {
            beforeEach(() => {
                Environment.CIRCLE_PULL_REQUEST_ID = '';
            });

            itMakesRequestWithCorrectParams();
        });
    });
});
