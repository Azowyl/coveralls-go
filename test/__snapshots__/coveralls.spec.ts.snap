// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Coveralls submitFromLcov with PR circleci environment makes request to coveralls with correct params 1`] = `
[
  "{"service_job_id":"1","service_name":"circleci","repo_token":"token","service_pull_request":"123","commit_sha":"","source_files":[{"name":"test/fixtures/repo/src/covered.ts","source_digest":"sourceDigest","coverage":[null,1,1,null,0,null,null,null,null,null],"branches":[1,0,0,1,2,1,0,1,2,1,1,0]},{"name":"test/fixtures/repo/src/uncovered.ts","source_digest":"sourceDigest","coverage":[null,0,null,null,null,null],"branches":[]}],"git":{"head":{"id":"0d15c65a92b7c3642883a528da8dd4841accde21","author_name":"test","committer_name":"'facugerstner_29@hotmail.com'\\n","message":"add missing snapshot\\n\\n"},"branch":"main\\n","remotes":[{"name":"origin","url":"https://github.com/Azowyl/coveralls-go.git\\n"}]}}",
]
`;

exports[`Coveralls submitFromLcov with no PR circleci environment makes request to coveralls with correct params 1`] = `
[
  "{"service_job_id":"1","service_name":"circleci","repo_token":"token","service_pull_request":"","commit_sha":"","source_files":[{"name":"test/fixtures/repo/src/covered.ts","source_digest":"sourceDigest","coverage":[null,1,1,null,0,null,null,null,null,null],"branches":[1,0,0,1,2,1,0,1,2,1,1,0]},{"name":"test/fixtures/repo/src/uncovered.ts","source_digest":"sourceDigest","coverage":[null,0,null,null,null,null],"branches":[]}],"git":{"head":{"id":"0d15c65a92b7c3642883a528da8dd4841accde21","author_name":"test","committer_name":"'facugerstner_29@hotmail.com'\\n","message":"add missing snapshot\\n\\n"},"branch":"main\\n","remotes":[{"name":"origin","url":"https://github.com/Azowyl/coveralls-go.git\\n"}]}}",
]
`;
